test/plugins/test-trace-http.ts

```diff
@@ -13,357 +13,420 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-'use strict';
 
-import { Constants } from '../../src/constants';
-import { TraceLabels } from '../../src/trace-labels';
+import * as assert from 'assert';
+import {EventEmitter} from 'events';
+import * as fs from 'fs';
+import * as httpModule from 'http';
+import * as httpsModule from 'https';
+import * as path from 'path';
+import * as semver from 'semver';
+import * as stream from 'stream';
+import {URL} from 'url';
 
-var common = require('./common'/*.js*/);
-var semver = require('semver');
-var stream = require('stream');
+import {TraceSpan} from '../../src/trace';
+import * as trace from '../trace';
+import {ASSERT_SPAN_TIME_TOLERANCE_MS, assertSpanDuration, DEFAULT_SPAN_DURATION} from '../utils';
+import {Express4} from '../web-frameworks/express';
 
-require('../../..').start({
-  projectId: '0',
-  samplingRate: 0
-});
+// This type describes (http|https).(get|request).
+type HttpRequest =
+    (options: string|httpModule.RequestOptions|httpsModule.RequestOptions|URL,
+     callback?: (res: httpModule.IncomingMessage) => void) =>
+        httpModule.ClientRequest;
 
-var assert = require('assert');
-
-describe('test-trace-http', function() {
-  var http;
-  var server;
-  before(function() {
-    http = require('http');
+/**
+ * A class that represents a convenience object that allows us to await
+ * http requests. This is done in lieu of promisifying http.request/http.get,
+ * because they already have a meaningful return value.
+ */
+class WaitForResponse {
+  private resolve!: (value: string) => void;
```

Comment:
Why do you need `!` here and below?

Reply:
TS doesn't know that the callback passed to the Promise constructor will be invoked synchronously (sound familiar? :))

Reply:
Now I understand. Wish TS understands how Promise works...